### DISCUSSION TITLE
QAInference with answer having a score

### DISCUSSION THREAD

**Discussion by xxx24xxx** [deepjavalibrary/djl#3554] [Q&A]
Regarding this Bert QAInference example: https://github.com/deepjavalibrary/djl/blob/master/examples/src/main/java/ai/djl/examples/inference/nlp/BertQaInference.java

Is it somehow possible that the answer also returns the score?
Like in the [transformer pipeline](https://huggingface.co/docs/transformers/tasks/question_answering)?
`{'score': 0.28, 'start': 10, 'end': 95, 'answer': 'that is the answer'}`

**Comment by frankfliu** [comment]
Ideally, the QuestionAnsweringTranslator should take `QAInput` and output `QAOutput`. You can implement your own Translator to output score for now.

To keep backward compatibility, we can implement a hacky solution to still return String in a json encoded format. Will push a PR soon.

**Comment by frankfliu** [comment]
@xxx24xxx 
See PR: https://github.com/deepjavalibrary/djl/pull/3555

**Reply by xxx24xxx** [reply]
Awesome! That's exactly what I'm talking about.
